src/tests/system/tests/test_gpo.py

```diff
@@ -0,0 +1,1056 @@
+"""
+SSSD GPO tests
+
+:requirement: IDM-SSSD-REQ: Group policy host based access control
+
+The following features and parameters are not covered:
+- ad_gpo_cache_timeout
+- ad_gpo_map_network
+- ad_gpo_map_batch
+- ad_gpo_map_service
+- ad_gpo_map_permit
+- ad_gpo_map_deny
+"""
+
+from __future__ import annotations
+
+import pytest
+
+from sssd_test_framework.roles.ad import AD
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["su", "ssh"])
+@pytest.mark.topology(KnownTopology.AD)
+def test_gpo__is_set_to_enforcing_and_users_are_allowed(client: Client, ad: AD, method: str):
+    """
+    :title: Group policy object host base access control is set to enforcing and users are allowed
+    :description:
+        When ad_gpo_access_control is set to enforcing, user access is managed and enforced by the policy.
+        Unlisted users are denied access.
+    :setup:
+        1. Create the following users; 'user1', 'user2'
+        2. Create the following groups; 'group' with member 'user2'
+        3. Create and link the GPO 'site policy' and add 'user1', group and 'Domain Admins' to
+           SeInteractiveLogonRight key.
+        4. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing'
+        5. Start SSSD
+    :steps:
+        1. Authenticate as 'user1' and 'user2'
+    :expectedresults:
+        1. Users authentication is successful
```

Comment:
I would suggest to add a third user which is not member of `group` and not added to to policy to make sure that access is not allowed unconditionally but that there is a real access control happening.

Reply:
This is covered in 'test_gpo__is_set_to_enforcing_and_users_are_not_in_policy', we have the policy with "Domain Admins' and attempt to login as 'user', any reference to 'user' with no appended numbers is a "regular"  user not in the policy.

Reply:
Hi,

imo this splitting is a level too deep. The reason to configure access control is typically to allow one set of users while rejecting a different set. Imo a single test for access control should demonstrate that at least some from the allowed set are actually allowed while at least some from the reject set are actually rejected. If only one side is tested how can I be sure the given configuration works as expected for the other side as well. And with individual two tests we would have the extra effort to make sure the setup for both will always be exactly the same.

bye,
Sumit

Reply:
I actually agree entirely, I'll go ahead and rework all the tests to have a positive and negative.

Reply:
After thinking about it some more, I don't think log checking is necessary here. I do want to improve the test framework for other tests, though. I'll add that a TODO for later. Are you okay with no log checking for this test?